### ISSUE TITLE
[retire] Retire from portfolio - dev implementation

### ISSUE THREAD

**Issue by jabby09** [KlimaDAO/klimadao#1106]
Related design task #1087 

See [Retirement Flow (from portfolio)](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-10822&t=ScV4oxm9NsoeqA4z-4) design.

### User journey and requirements:

- [x] Note desktop and mobile versions
- [x] From Portfolio page, click RETIRE button on project card in your portfolio of assets
- [x] Takes user to Retire form page, design [here](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-10835&t=ScV4oxm9NsoeqA4z-4)
    - Ignore for now the translation option in the header
    - Need to check on the need for the "< BACK TO PORTFOLIO" breadcrumb
    - Need to check on the Asset Details icon
- [x] User completes form and clicks RETIRE CARBON
- [x] Confirm Retirement modal, APPROVE screen, design [here](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-10980&t=ScV4oxm9NsoeqA4z-4)
- [x] Change intro text from "Please approve the transaction" to:
> You are about to retire a carbon asset.
> 
> The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.
> 
> Verify all information is correct and click 'approve' to continue.

- [x] User clicks APPROVE
- [x] Confirm Retirement modal, SUBMIT screen
- [x] Change intro text from "Please submit the transaction" to:

> The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.
> 
> To finalize your retirement, verify all information is correct and then click 'submit' below.

- [x] Processing Retirement spinner while waiting on transaction to complete, design [here](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-10991&t=ScV4oxm9NsoeqA4z-4)
- [x] Payment Successful screen, design [here](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-11601&t=ScV4oxm9NsoeqA4z-4)
- [x] RETIRE MORE CARBON button takes user back to /portfolio page
- [x] VIEW AND SHARE CERTIFICATE  button takes user to retirement receipt page

**_Notes on retirement receipt and pdf pages_**

- The interim solution to support is a copy, paste, tweak of existing page design, as per #1087 
- The stretch goal is to implement the new design of the [receipt page](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-10996&t=ScV4oxm9NsoeqA4z-4) and [download pdf](https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=247-11153&t=ScV4oxm9NsoeqA4z-4)

**Comment by jabby09** [comment]
@Atmosfearful please add any other technical requirements or details for @psparacino

**Comment by Atmosfearful** [comment]
This will start after the plumbing in #1097

**Comment by jabby09** [comment]
@ShimonD-KlimaDAO this user flow is ready for initial QA

Note there was a related "plumbing" ticket (#1097) and the PR for all these changes is #1145 

PR link address to test from: https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/

Other testing notes:
- Currently the RETIRE button link is not hooked up (this is part of the plumbing work in the PR)
- You can just input the url to test the retire flow, i.e. `/portfolio/<token address here>/retire`
- To get the token address you can right click copy the link destination on the pop up notification redirect modal and the token address will be in there

**Comment by jabby09** [comment]
@psparacino (cc @ShimonD-KlimaDAO) initial testing feedback...

- [ ] Default banner images are not using the darkened version so the white text is not readable. See a [project details](https://www.carbonmark.com/projects/VCS-1140-2015) page to see examples of the darkened default images. 
<img width="733" alt="Screenshot 2023-05-23 at 9 42 25 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/1a10b4ef-35b0-4cd7-ac31-dcdafdd7c9d4">

- [ ] This copy does not match design, please change to:
> Be careful not to include any sensitive personal information (such as an email address) in your retirement name or message. The information you enter here cannot be edited once it is submitted and will permanently exist on a public blockchain.
<img width="508" alt="Screenshot 2023-05-23 at 9 43 00 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/7bd426e7-f9b6-4d6d-b6fc-807c9bcaac15">

- [ ] Only the amount of tonnes is a required field so it needs the * label as per the design
![Screenshot 2023-05-23 at 12 08 07 PM](https://github.com/KlimaDAO/klimadao/assets/103535195/b5228577-51fa-4c0f-b05d-49f591d25341)

- [ ] When tonnes are entered, the RETIRE CARBON button should be activated
<img width="401" alt="Screenshot 2023-05-23 at 9 44 05 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/86b32557-30f0-480d-8dcc-bbc6fb39e601">

- [ ] Beneficiary and retirement message are not required fields. Currently I had to enter these to get the RETIRE CARBON button to activate
<img width="389" alt="Screenshot 2023-05-23 at 9 44 49 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/75ce4582-ca3c-48f2-8cdf-3d2bb5f2e016">

- [ ] As you noted in discord, Confirm Retirement modal is not following the design implementation in other parts of the site, this needs to change to be consistent

- [ ] Please change the copy on the APPROVE screen to:

> You are about to retire a carbon asset.
> 
> The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.
> 
> Verify all information is correct and click 'approve' to continue.
<img width="561" alt="Screenshot 2023-05-23 at 11 53 29 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/d0814a9c-dc04-4589-87aa-3a0f0b86c0f5">

- [ ] Please change the copy of the SUBMIT screen to:

> The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.
> 
> To finalize your retirement, verify all information is correct and then click 'submit' below.
<img width="550" alt="Screenshot 2023-05-23 at 11 54 40 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/db8f4efe-5179-4ec8-8269-96da72488ab6">

- [ ] Processing retirement spinner modal is not matching the design dimensions, it appears much longer width
<img width="648" alt="Screenshot 2023-05-23 at 9 47 06 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/df925fd4-5566-4efb-b04a-1b7f99e89e70">

- [ ] View transaction on polygonscan is not linking to the polygon
<img width="559" alt="Screenshot 2023-05-23 at 12 00 39 PM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/5c5452b0-1a6f-410d-aafa-3383d6338035">
scan transaction, it is linking to the retirement receipt
